src/geocoder.jl

```diff
@@ -8,6 +8,65 @@ Column names in the geonames dumpfile (from http://download.geonames.org/export/
 const COLUMNS = [:geonameid, :name, :asciiname, :alternatenames, :latitude, :longitude, 
                 :feature_class, :feature_code, :country_code, :cc2, :admin1_code, :admin2_code, 
                 :admin3_code, :admin4_code, :population, :elevation, :dem, :timezone, :modification_date]
+
+const COLUMN_TYPE = Dict(
+    :geonameid => Int, 
+    :name => String, 
+    :asciiname => String, 
+    :alternatenames => String, 
+    :latitude => Float64, 
+    :longitude => Float64, 
+    :feature_class => String, 
+    :feature_code => String, 
+    :country_code => String, 
+    :cc2 => String, 
+    :admin1_code => String, 
+    :admin2_code => String, 
+    :admin3_code => String, 
+    :admin4_code => String, 
+    :population => Int, 
+    :elevation => Int,
+    :dem => Int, 
+    :timezone => String, 
+    :modification_date => String
+)
+
+function _extract_column_values(row::DataFrameRow, headers)::Vector{Any}
+    values = Vector{Any}(undef, length(headers))
+    
+    for (i, header) ∈ enumerate(headers)
+        value = getproperty(row, header)
+        type  = COLUMN_TYPE[header]
+
+        if ismissing(value)
+            value = type == String ? "" : 0
+        elseif type == String
+            value = string(value)
+        end
+
+        values[i] = value
+    end
+
+    values
+end
+
+function _split_latlon_and_info(data::AbstractDataFrame)
+    info_headers = Tuple(filter(x -> x ∉ [:latitude, :longitude], Symbol.(names(data))))
+    
+    n = nrow(data)
+    points = Array{Float64}(undef, 2, n)
+    info   = _extract_column_values(data[1, :], info_headers)
```

Comment:
Put this line inside the `Tuple` on the following line to avoid two variables named `info` with different meanings in the same function.

Reply:
Alrighty addressed below!